pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -197,30 +200,70 @@ func (c *Cluster) createSelfSignedCert() (bool, error) {
 	return false, nil
 }
 
+// CephVersionsNewCephDashboardCommand check if Ceph versions have the latest Ceph dashboard command
+func CephVersionsNewCephDashboardCommand(c *client.ClusterInfo) bool {
+	if (c.CephVersion.IsNautilus() && c.CephVersion.IsAtLeast(cephver.CephVersion{Major: 14, Minor: 2, Extra: 17})) ||
+		(c.CephVersion.IsOctopus() && c.CephVersion.IsAtLeast(cephver.CephVersion{Major: 15, Minor: 2, Extra: 10})) ||
+		c.CephVersion.IsAtLeastPacific() || c.CephVersion.IsAtLeastQuincy() {
+		return true
+	}
+	return false
+}
+
 func (c *Cluster) setLoginCredentials(password string) error {
 	// Set the login credentials. Write the command/args to the debug log so we don't write the password by default to the log.
 	logger.Infof("setting ceph dashboard %q login creds", dashboardUsername)
 
-	// retry a few times in the case that the mgr module is not ready to accept commands
-	_, err := client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
-		args := []string{"dashboard", "set-login-credentials", dashboardUsername, password}
-		cmd := client.NewCephCommand(c.context, c.clusterInfo, args)
-		output, err := cmd.RunWithTimeout(client.CmdExecuteTimeout)
-		return "set dashboard creds", output, err
-	}, c.exitCode, 5, invalidArgErrorCode, dashboardInitWaitTime)
-	if err != nil {
-		return errors.Wrapf(err, "failed to set login creds on mgr")
+	cephVersion := CephVersionsNewCephDashboardCommand(c.clusterInfo)
+	// for latest Ceph versions
+	if cephVersion {
+		// Generate a temp file
+		file, err := ioutil.TempFile("", "")
+		if err != nil {
+			return errors.Wrap(err, "failed to generate temp file")
+		}
+		err = ioutil.WriteFile(file.Name(), []byte(password), 0444)
```

Comment:
Generally, I think 0440 is a more secure option, but I think the Rook operator runs as the `root` user anyway, so this might be irrelevant.

```suggestion
		err = ioutil.WriteFile(file.Name(), []byte(password), 0440)
```

Reply:
Done.